core/domain/state_domain.py

```diff
@@ -1426,7 +1426,14 @@ def validate(self):
 
 
 class WrittenTranslation(python_utils.OBJECT):
-    """Value object representing a written translation for a content."""
+    """Value object representing a written translation for a content.
+
+    Here, "content" could mean a string or a list of strings. The latter arises,
+    for example, in the case where we are checking for equality of a learner's
+    answer against a given set of strings. In such cases, the number of strings
+    in the translation of the original object may not be the same as the number
+    of strings in the original object.
```

Comment:
I think that would be more complicated, since the two types have different meanings. In the "string" type, that exact string must be translated and the translation must correspond to the string. In the "list" type, the entire list must be translated, and the correspondence between items in that list is not one-to-one -- indeed, the source list and target list could even have different lengths! 

So, I think that using a list for all the cases is likely to lead to more confusion because it implies a false equivalence between the two. Would you agree?

Reply:
Ah ok, we can leave it as it is then. Though just curious, when does source and target list differ? Is it some text input rules where the rule can be represented as a single string in some languages?

Reply:
Yes, that's right, but there are other cases too. Imagine e.g. if a language had two different words for "numerator". Or in Hinglish where we might want to accept both the English and devanagari versions.

Reply:
Yeah ok, makes sense.